src/NServiceBus.Core/Sagas/SagaMetadata.cs

```diff
@@ -378,13 +383,15 @@ static void ValidateMapping<TMessage>(Expression<Func<TMessage, object>> message
 
                 var propertyInfo = memberExpr.Member as PropertyInfo;
 
-                const string message = "Message properties mapped to the saga id needs to be of type Guid, please change property {0} on message {1} to a Guid";
+                const string message = "When mapping a message to a saga, the member type on the message and the saga property must match. {0}.{1} is of type {2} and {3}.{4} is of type {5}.";
 
                 if (propertyInfo != null)
                 {
                     if (propertyInfo.PropertyType != sagaProp.PropertyType)
                     {
-                        throw new Exception(string.Format(message, propertyInfo.Name, typeof(TMessage).Name));
+                        throw new InvalidOperationException(string.Format(message,
```

Comment:
the difference between InvalidOperationException and Exception is cosmetic when there is no intention for the user to catch it